odoo/orm/decorators.py

```diff
@@ -40,12 +35,33 @@
 #  - method._api: decorator function, used for re-applying decorator
 #
 
-def attrsetter(attr, value):
+def attrsetter(attr, value) -> Decorator:
     """ Return a function that sets ``attr`` on its argument and returns it. """
-    return lambda method: setattr(method, attr, value) or method
+    return lambda method: setattr(method, attr, value) or method  # type: ignore
 
 
-def constrains(*args: str) -> Callable[[T], T]:
+def propagate(method1: T | None, method2: T) -> T:
+    """ Propagate decorators from ``method1`` to ``method2``, and return the
+        resulting method.
+    """
+    if method1:
+        for attr in ('_returns',):
+            if hasattr(method1, attr) and not hasattr(method2, attr):
+                setattr(method2, attr, getattr(method1, attr))
+    return method2
+
+
+@typing.overload
+def constrains(func: Callable[[BaseModel], Collection[str]], /) -> Decorator:
+    ...
+
+
+@typing.overload
+def constrains(*args: str) -> Decorator:
+    ...
+
+
+def constrains(*args) -> Callable[[T], T]:
```

Comment:
This type signature is not consistent with the ones above.
```suggestion
def constrains(*args) -> Decorator:
```